src/Auxiliary/Aux_Check_Parameter.cpp

```diff
@@ -650,14 +655,21 @@ void Aux_Check_Parameter()
 #   if ( RSOLVER_RESCUE != NONE  &&  RSOLVER_RESCUE != ROE  &&  RSOLVER_RESCUE != HLLE  &&  RSOLVER_RESCUE != HLLD )
 #     error : ERROR : unsupported RSOLVER_RESCUE for MHD (ROE/HLLE/HLLD) !!
 #   endif
+#  elif SRHD 
+#   if ( defined RSOLVER  &&  RSOLVER != HLLC  &&  RSOLVER != HLLE )
+#     error : ERROR : unsupported Riemann solver for SRHD (HLLC/HLLE) !!
+#   endif
+#   if ( defined FLU_SCHEME  &&  FLU_SCHEME != MHM_RP && FLU_SCHEME != MHM )
+#     error : ERROR : unsupported FLU_SCHEME for SRHD (MHM_RP/MHM) !!
+#   endif
```

Comment:
How about separating this section into an individual `#if` section? Also, we need to add more checks for `SRHD`,  e.g. `GRAVITY` should be `false`, `MHD` should be `false`, etc.